src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Renderer3d.kt

```diff
@@ -0,0 +1,303 @@
+package com.chattriggers.ctjs.minecraft.libs.renderer
+
+import com.chattriggers.ctjs.minecraft.wrappers.Client
+import com.chattriggers.ctjs.minecraft.wrappers.Settings
+import com.chattriggers.ctjs.utils.get
+import com.chattriggers.ctjs.utils.vec.Vec3f
+import com.mojang.blaze3d.systems.RenderSystem
+import gg.essential.elementa.dsl.component1
+import gg.essential.elementa.dsl.component2
+import gg.essential.elementa.dsl.component3
+import gg.essential.elementa.dsl.component4
+import gg.essential.universal.UGraphics
+import gg.essential.universal.UMatrixStack
+import net.minecraft.client.MinecraftClient
+import net.minecraft.client.font.TextRenderer
+import net.minecraft.client.render.Tessellator
+import net.minecraft.client.render.VertexFormat
+import org.joml.Vector3f
+import org.mozilla.javascript.NativeObject
+import java.awt.Color
+import kotlin.math.sqrt
+
+object Renderer3d {
+    private var firstVertex = true
+    private var began = false
+
+    private val tessellator = Tessellator.getInstance()
+    private val worldRenderer = UGraphics(tessellator.buffer)
+
+    /**
+     * Begin drawing with the world renderer
+     *
+     * @param drawMode the GL draw mode
+     * @param vertexFormat The [VertexFormat] to use for drawing
+     * @return the Renderer to allow for method chaining
+     * @see com.chattriggers.ctjs.minecraft.libs.renderer.Shape.setDrawMode
+     */
+    @JvmStatic
+    @JvmOverloads
+    fun begin(
+        drawMode: Renderer.DrawMode = Renderer.DrawMode.QUADS,
+        vertexFormat: Renderer.VertexFormat = Renderer.VertexFormat.POSITION,
+    ) = apply {
+        Renderer.pushMatrix()
+        Renderer.enableBlend()
+        Renderer.tryBlendFuncSeparate(770, 771, 1, 0)
+
+        worldRenderer.beginWithDefaultShader(drawMode.toUC(), vertexFormat.toMC())
+
+        firstVertex = true
+        began = true
+    }
+
+    /**
+     * Sets a new vertex in the Tessellator.
+     *
+     * @param x the x position
+     * @param y the y position
+     * @param z the z position
+     * @return the Tessellator to allow for method chaining
+     */
+    @JvmStatic
+    fun pos(x: Float, y: Float, z: Float) = pos(x.toDouble(), y.toDouble(), z.toDouble())
```

Comment:
Is this needed? afaik all numbers in Rhino are `Double`s anyways. If it is needed it should be `Number` instead so it can take anything. And this comment applies to all functions here that take `Float`s.

Reply:
No, it's not needed, I'll delete that.